tangelo/linq/qpu_connection/ibm_connection.py

```diff
@@ -0,0 +1,195 @@
+"""
+    Wrappers around Qiskit runtime API, to manage quantum experiments run with IBM Cloud or IBM quantum
+    from Tangelo.
+"""
+
+import os
+
+from tangelo.linq.translator import translate_operator, translate_circuit
+from tangelo.linq.qpu_connection.qpu_connection import QpuConnection
+
+try:
+    from qiskit.providers.jobstatus import JobStatus
+    from qiskit_ibm_runtime import QiskitRuntimeService, Sampler, Estimator, Session, Options, \
+        SamplerResult, EstimatorResult
+    is_qiskit_installed = True
+except ModuleNotFoundError:
+    is_qiskit_installed = False
+
+
+class IBMConnection(QpuConnection):
+    """ Wrapper around IBM Qiskit runtime API to facilitate job submission from Tangelo """
+
+    def __init__(self, ibm_quantum_token=None):
+
+        if not is_qiskit_installed:
+            raise ModuleNotFoundError("Both qiskit and qiskit_ibm_runtime need to be installed.")
+
+        self.api_key = ibm_quantum_token if ibm_quantum_token else os.getenv("IBM_TOKEN", None)
+        self.service = self._login()
+        self.jobs = dict()
+        self.jobs_results = dict()
+
+    def _login(self):
+        """ Attempt to connect to the service. Fails if environment variable IBM_TOKEN
+            has not been set to a correct value.
+        """
+        if not self.api_key:
+            raise RuntimeError(f"Please provide IBM_TOKEN (as environment variable or at instantiation of connection.")
+        try:
+            return QiskitRuntimeService(channel="ibm_quantum", token=self.api_key)
+        except Exception as err:
+            raise RuntimeError(f"{err}")
+
+    def get_backend_info(self):
+        """ Return configuration information for each device found on the service """
+        return {b.name: b.configuration() for b in self.service.backends()}
+
+    def job_submit(self, program, backend_name, circuits, operators=None, n_shots=10**4, runtime_options=None):
+        """ Submit job, return job ID.
+
+        Args:
+            program (str): name of available qiskit-runtime program (e.g sampler, estimator currently)
+            backend_name (str): name of a qiskit backend
+            circuits (Circuit | List[Circuit]): Tangelo circuit(s)
+            operators (QubitOperator | List[QubitOperator]) : Optional, qubit operators for computing expectation values
+            n_shots (int): Optional, number of shots to use on the target backend
```

Comment:
Personally, I would let `n_shots` to be provided by the user. 10**4 is kind of in the low side. Also, is there a difference between `n_shots` using the sampler and the estimator (and specify it in the doc)? I would assume that the real number of shots for estimator is `n_shots * len(operators)`.

Reply:
@AlexandreF-1qbit Yeah I think the number of shots should not be optional, as it directly relates to the amount of time / computation someone is going to be billed, and the accuracy of results. The user should have their eyes on it.

qiskit-runtime is still seeing a lot of changes, and the documentation and examples are a bit all over the place. I think Estimator may also do things behind the scenes (QWC, others), as each of these operators in `[operators]` can be a Hamiltonian.

I think the right course of action is requiring a number of shots from the user, and letting IBM Quantum do its thing.

Reply:
All good with me!